zcutil/make-release.py

```diff
@@ -209,29 +222,20 @@ def verify_version(release, releaseprev, hotfix):
 
 
 @phase('Initializing git.')
-def initialize_git(release, hotfix):
+def initialize_git(revision, release, hotfix):
     junk = sh_out('git', 'status', '--porcelain')
     if junk.strip():
         raise SystemExit('There are uncommitted changes:\n' + junk)
 
-    branch = sh_out('git', 'rev-parse', '--abbrev-ref', 'HEAD').strip()
-    if hotfix:
-        expected = 'hotfix-' + release.vtext
-    else:
-        expected = 'master'
-    if branch != expected:
-        raise SystemExit(
-            "Expected branch {!r}, found branch {!r}".format(
-                expected, branch,
-            ),
-        )
-
-    logging.info('Pulling to latest master.')
-    sh_log('git', 'pull', '--ff-only')
-
     branch = 'release-' + release.vtext
-    logging.info('Creating release branch: %r', branch)
-    sh_log('git', 'checkout', '-b', branch)
+    logging.info(
+        'Creating release branch {} from revision {}.'
+        .format(
+            branch, 
+            revision.value
+        )
+    )
+    sh_log('git', 'checkout', '-b', branch, revision.value)
```

Comment:
Don't we want this to succeed if the branch already exists and is at that commit?

Reply:
I think it's a useful property to have that our release branches are always created by the release script, instead of the hybrid approach that we have right now where hotfix branches are created manually but normal release branches are created automatically?